### ISSUE TITLE
Billing > Diagnosis pointers

### ISSUE THREAD

**Issue by rumsant** [shihjay2/nosh2#46]
This may simply be due to a lack of understanding on my part because I don't deal with third party payers, so I'm not terribly fluent in their billing protocols.

I can add a a Procedure Code for an encounter and use the Diagnosis Pointer to associate with a diagnosis.

However, if I attempt to add a second Procedure Code, I get an error as follows:
"QueryException in Connection.php line 770:
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY' (SQL: insert into `billing_core` (`billing_core_id`, `cpt`, `cpt_charge`, `unit`, `modifier`, `dos_f`, `dos_t`, `icd_pointer`, `eid`, `practice_id`, `pid`) values (, 72040, 150, 1, , 04/03/2017, 04/03/2017, AB, 1, 1, 1))"

Am I doing something that is legitimately not allowed? Or is this an error?

**Comment by shihjay2** [comment]
You're doing it right.  Fixed in commit 3583d11880efeae5de6a43cfe460e8146e895b01

**Comment by rumsant** [comment]
Added a second encounter with a dummy patient. In the billing section for that encounter, I added a procedure code. When I clicked on Diagnosis Pointers, I was greeted with the same diagnoses that I had entered in the first encounter. I am assuming that is how it is supposed to work (i.e. that even though I hadn't entered those diagnoses for the current encounter, they persist for that patient for all encounters).

The diagnosis code that I entered for the current encounter did not show up in the Diagnosis Pointers section, even though it had been saved earlier in the same encounter.

I chose one of the Diagnosis Pointers that was present and clicked Save for the Procedure. I was given the following error:
"QueryException in Connection.php line 770:
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY' (SQL: insert into `billing_core` (`cpt`, `cpt_charge`, `unit`, `modifier`, `dos_f`, `dos_t`, `icd_pointer`, `eid`, `practice_id`, `pid`) values (99201, 60, 1, , 04/03/2017, 04/03/2017, A, 1, 1, 1))"

I don't know if this is happening because both encounters are occurring on the same day, or if it is an error.

**Comment by shihjay2** [comment]
Do you have cookies enabled in your browser and are you clicking the back button.  It sounds like your sessions are somehow corrupted?  You can confirm on the left hand side where there is an encounter number (Active Encounter ##).  If it's not the encounter that you're in even though you think you're in it, please let me know.  I not able to replicate this on my systems so far.

**Comment by shihjay2** [comment]
I'm beginning to suspect this is related to Vagrant implementations and file permissions (specifically with sessions handling).  I'll be posting a fix once I figure out what is going on.

**Comment by rumsant** [comment]
Yes, I do have cookies enabled on the browser. 
I believe I'm using the back button, but I will pay closer attention to that in the future. 
I will check the active encounter# when I have access tomorrow, but I believe it did match up correctly.  The other encounter had been signed, so I don't think I could have been attached to it.

**Comment by shihjay2** [comment]
in Vagrantfile can you substitute
`config.vm.synced_folder "", "/vagrant"`
for
`config.vm.synced_folder "", "/vagrant", :mount_options => ["dmode=777","fmode=666"]`

reload Vagrant and see if you're still having the same issues.

**Comment by rumsant** [comment]
I will let you know when I have access to the system tomorrow.

**Comment by rumsant** [comment]
I've modified the vagrantfile as directed. The proper ICD codes are now propagating in the Diagnosis Pointers dropdown menu.
However, when I try to save a Procedure Code for Encounter, I am taken to the following error page:
"QueryException in Connection.php line 770:
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY' (SQL: insert into `billing_core` (`cpt`, `cpt_charge`, `unit`, `modifier`, `dos_f`, `dos_t`, `icd_pointer`, `eid`, `practice_id`, `pid`) values (99201, 60, 1, , 04/03/2017, 04/03/2017, AB, 2, 1, 1))"

**Comment by shihjay2** [comment]
What is the URL that you see in the browser when the error is show?

**Comment by rumsant** [comment]
https://192.168.1.121/nosh/chart_action/billing_core/save/0/billing_core_id

**Comment by shihjay2** [comment]
Fixed in commit 8eb4185a70fd6dcc2ef80d0bf578fd9787f1cde2

**Comment by rumsant** [comment]
Update fixed ability to save Procedure Code for the encounter.

I then attempted to sign the encounter and was alerted that that there were missing items for billing.
I thought that perhaps I needed to go to the "Who" tab in the billing window. I went there and clicked the "Save" button and was taken to a page with the following error:
"QueryException in Connection.php line 770:
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY' (SQL: insert into `billing` (`eid`, `pid`, `insurance_id_1`, `insurance_id_2`, `bill_complex`) values (2, 1, 0, , ))"
